workflow/artifacts/oss/oss.go

```diff
@@ -148,6 +152,53 @@ func (ossDriver *ArtifactDriver) ListObjects(artifact *wfv1.Artifact) ([]string,
 	return files, err
 }
 
+func (ossDriver *ArtifactDriver) setBucketLifecycleRule(client *oss.Client, artifact *wfv1.Artifact) error {
+	if artifact.OSS.LifecycleRule.MarkInfrequentAccessAfterDays == nil && artifact.OSS.LifecycleRule.MarkDeletionAfterDays == nil {
+		return nil
+	}
+	var markInfrequentAccessAfterDays int
+	var markDeletionAfterDays int
+	if artifact.OSS.LifecycleRule.MarkInfrequentAccessAfterDays != nil {
+		markInfrequentAccessAfterDays = int(*artifact.OSS.LifecycleRule.MarkInfrequentAccessAfterDays)
+	}
+	if artifact.OSS.LifecycleRule.MarkDeletionAfterDays != nil {
+		markDeletionAfterDays = int(*artifact.OSS.LifecycleRule.MarkDeletionAfterDays)
+	}
+	if markInfrequentAccessAfterDays > markDeletionAfterDays {
+		return fmt.Errorf("markInfrequentAccessAfterDays cannot be large than markDeletionAfterDays")
+	}
+
+	// Set expiration rule.
+	expirationRule := oss.BuildLifecycleRuleByDays("expiration-rule", artifact.OSS.Key, true, markInfrequentAccessAfterDays)
+	// Automatically delete the expired delete tag so we don't have to manage it ourselves.
+	deleteMark := true
```

Comment:
suggestion - `pointer.BoolPtr(true` and inline

Reply:
Great suggestion! Updated.